frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -1118,9 +1123,13 @@ function ReaderDictionary:downloadDictionary(dict, download_location, continue)
         return false
     end
 
-    local ok, error = Device:unpackArchive(download_location, self.data_dir)
+    -- stable target directory is needed so we can look through the folder later
+    local dict_path = self.data_dir .. "/" .. dict.name
+    util.makePath(dict_path)
+    local ok, error = Device:unpackArchive(download_location, dict_path, true)
```

Comment:
Dunno how problematic or not it is to make a directory named as the name= of what's in our dictionaries.lua:
- with their possible non-ASCII content - but I guess everybody is already able to create file and directories with books on all devices
- the fact that it seems we have 7 duplicated names - but then, before, just untaring and keeping the subdirectories in the downloaded tar.gz, we could already have duplicated directories.

Reply:
I figured it should be okay since there's nothing preventing both of those things with the current code either.
Maybe we could make those dict names in the list unique? Not sure if it's worth the effort.